cmake/CudaRdcUtils.cmake

```diff
@@ -141,6 +147,20 @@ define_property(TARGET PROPERTY CUDA_RDC_OBJECT_LIBRARY
   FULL_DOCS "Name of the object (without nvlink step) library corresponding to this cuda library"
 )
 
+if(CMAKE_SYSTEM_NAME STREQUAL "Linux")
+  # This is necessary on platform that default to calling ld with --as-needed
+  # (we are looking at you Ubuntu) but also in case user as explicitly using it.
+  set(CMAKE_CXX_LINK_LIBRARY_USING_rdc_no_as_needed_SUPPORTED TRUE CACHE INTERNAL "")
+  set(CMAKE_CXX_LINK_LIBRARY_USING_rdc_no_as_needed
```

Comment:
Can you add documentation linking to https://cmake.org/cmake/help/latest/variable/CMAKE_LINK_LIBRARY_USING_FEATURE.html ?This also bumps the minimum CMake version for linux+cuda+vecgeom to 3.24.

Reply:
Um honestly I think the best thing would be a "cmake minimum" at the top of the file. Since:
> CMake also manages a new entry for scripts loaded by [include()](https://cmake.org/cmake/help/latest/command/include.html#command:include) and [find_package()](https://cmake.org/cmake/help/latest/command/find_package.html#command:find_package) commands except when invoked with the NO_POLICY_SCOPE option (see also policy [CMP0011](https://cmake.org/cmake/help/latest/policy/CMP0011.html#policy:CMP0011))

then we could probably just put `cmake_policy(VERSION 3.24...3.31)` at the top of the file?

Reply:
I think that's a decent failure mode.